aws/resource_aws_instance.go

```diff
@@ -1528,6 +1543,46 @@ func readSecurityGroups(d *schema.ResourceData, instance *ec2.Instance, conn *ec
 	return nil
 }
 
+func readPasswordData(d *schema.ResourceData, instance *ec2.Instance, conn *ec2.EC2) error {
+	if !d.Get("get_password_data").(bool) {
+		if err := d.Set("password_data", nil); err != nil {
+			return err
+		}
+
+		return nil
+	}
+
+	log.Printf("[INFO] Reading password data for instance %s", d.Id())
+
+	timeout := time.Now().Add(15 * time.Minute)
```

Comment:
Nitpick: we have the `resource.Retry` function we regularly use for these type of retryable operations:

```go
var passwordData string

input := &ec2.GetPasswordDataInput{
  InstanceId: aws.String(instanceID),
}

err := resource.Retry(15 * time.Minute, func() *resource.RetryError {
  out, err := conn.GetPasswordData(input)

  if err != nil {
    return resource.NonRetryableError(err)
  }

  if resp.PasswordData == nil || *resp.PasswordData == "" {
    return resource.RetryableError(fmt.Errorf("Password data is blank for instance ID: %s", instanceID))
  }

  passwordData = strings.TrimSpace(*resp.PasswordData)
  return nil
})

if err != nil {
  return "", err
}

return passwordData, nil
```